### PR TITLE
Rs issue#8

### PR DESCRIPTION
**Pull request by Robbie-Smith**

### CODE REVIEW

**Comment by Dpalazzari** [comment]
After we fix the unsucessful check, should merge asap.

**Comment by jdconrad89** [comment]
@Robbie-Smith looks good! Thanks for fixing all the GitIgnore issues! great job on oauth! 

image

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -4,23 +4,24 @@
...
 
   context 'associations' do
+
     it { is_expected.to have_many :user_roles }
     it { is_expected.to have_many :roles }
     it { is_expected.to have_many :owner_user_projects }
-    it { is_expected.to have_many :owned_projects } 
+    it { is_expected.to have_many :owned_projects }
     it { is_expected.to have_many :funded_user_projects }
-    it { is_expected.to have_many :funded_projects }
```
Comment (kheppenstall) [inline spec/models/user_spec.rb]: Why are we deleting these have_many owned projects and funded projects?

Comment (kheppenstall) [inline spec/models/user_spec.rb]: @Robbie-Smith

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -17,13 +19,15 @@
   config.allow_http_connections_when_no_cassette = true
   config.cassette_library_dir = "spec/fixtures/vcr_cassettes"
   config.hook_into :webmock
-  config.filter_sensitive_data('<Client_id') do |interaction|
+  config.filter_sensitive_data('<client_id>') do |interaction|
     ENV['client_id']
   end
-  config.filter_sensitive_data('<Client_secret>') do |interaction|
+  config.filter_sensitive_data('<client_secret>') do |interaction|
```
Comment (kheppenstall) [inline spec/rails_helper.rb]: What sensitive data to we have for our API that we have to delete?

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.find_or_create_by_omniauth(oauth_data)
+    session[:user_id] = user.id
```
Comment (Dpalazzari) [inline app/controllers/sessions_controller.rb]: Should build the current_user helper method, too.

File: spec/factories/users.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 FactoryGirl.define do
   factory :user do
     name            { Faker::GameOfThrones.character + rand(1000).to_s }
+    slug { name.parameterize }
```
Comment (kheppenstall) [inline spec/factories/users.rb]: Let's go into the seed file and do the same thing here.

File: spec/models/user_spec.rb (hunk 1 above)
Comment (kheppenstall) [inline spec/models/user_spec.rb]: Nevermind on this one. It is all here it is just moved around.

File: spec/factories/users.rb (hunk 4 above)
Comment (Robbie-Smith) [inline spec/factories/users.rb]: Sounds good!

File: spec/rails_helper.rb (hunk 2 above)
Comment (Robbie-Smith) [inline spec/rails_helper.rb]: You're right, I'll remove that.

File: spec/requests/authentication_spec.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+describe 'Authentication' do
+  before do
+    oauth_setup
+    Rails.application.env_config['omniauth.auth'] = OmniAuth.config.mock_auth[:facebook]
+    get '/auth/facebook/callback'
```
Comment (Dpalazzari) [inline spec/requests/authentication_spec.rb]: Is this the equivalent of creating a session through the omniauth?

**Pull request merged**